app/models/journal_notification_mailer.rb

```diff
@@ -0,0 +1,86 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+class JournalNotificationMailer
+  class << self
+    def distinguish_journals(journal, send_notification)
+      if journal.journable_type == 'WorkPackage' && send_notification && journal.initial?
+        handle_create(journal.journable)
+      elsif journal.journable_type == 'WorkPackage' && send_notification
+        handle_update(journal)
+      end
+    end
+
+    def handle_create(work_package)
```

Comment:
Since this method is very work package specific I'd probably call it `handle_work_package_create`.

I don't know why this class is not yet responsible for journals of other classes than work packages, but it at least seems likely that it will in the future.